### PR TITLE
[ROCm][MLA] enable fp8 MLA decode on ROCm

### PR DESCRIPTION
**Pull request by gbyu-amd**
## Purpose
AITER MLA decode supports fp8 inputs now, so add the related scale arguments accordingly.

## Test Plan
The accuracy and performance of Deepseek-v3 are verified on MI355.

Server cmd:
```bash
#!/bin/bash
export VLLM_USE_V1=1
export SAFETENSORS_FAST_GPU=1
export VLLM_ROCM_USE_AITER=1
export VLLM_ROCM_USE_AITER_MOE=1
export VLLM_USE_TRITON_FLASH_ATTN=0
export NCCL_DEBUG=WARN
export VLLM_RPC_TIMEOUT=1800000
export VLLM_ROCM_USE_AITER_MHA=0
export VLLM_ROCM_USE_TRITON_ROPE=1
export VLLM_ROCM_USE_AITER_FUSION_SHARED_EXPERTS=1

vllm serve deepseek-ai/DeepSeek-V3 \
    --tensor-parallel-size 8 \
    --max-num-batched-tokens 32768 \
    --trust-remote-code \
    --no-enable-prefix-caching \
    --disable-log-requests \
    --gpu_memory_utilization 0.9 \
    --compilation-config '{"cudagraph_mode": "FULL_AND_PIECEWISE"}' \
    --block-size 128 \
    --async-scheduling \
```
Add `--kv-cache-dtype fp8` to enable fp8 MLA.

Benchmark cmd:
```bash
vllm bench serve \
            --host localhost \
            --port 8000 \
            --model deepseek-ai/DeepSeek-V3 \
            --dataset-name random \
            --random-input-len ${input_tokens} \
            --random-output-len ${output_tokens} \
            --max-concurrency ${max_concurrency} \
            --num-prompts ${num_prompts} \
            --percentile-metrics ttft,tpot,itl,e2el \
            --ignore-eos \
            --seed 123
```
lm_eval cmd:
```bash
lm_eval \
--model local-completions \
--tasks gsm8k \
--model_args model=deepseek-ai/DeepSeek-V3,base_url=http://127.0.0.1:8000/v1/completions \
--batch_size 100 
```

### Performance data
  | Input tokens | Output tokens | Max concurrency | Num prompts | Mean TTFT(ms) | Mean TPOT(ms) | Reqs/sec | QPM | Total token throughput | Throughput uplift ratio
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
BF16 | 8192 | 1024 | 64 | 128 | 7496.48 | 43.86 | 1.22 | 73.2 | 11238.26 | 100.00%
| |16384 | 1024 | 64 | 128 | 13235.75 | 65.22 | 0.8 | 48 | 13894.35 | 100.00%
FP8 | 8192 | 1024 | 64 | 128 | 7108.03 | 41.89 | 1.28 | 76.8 | 11782.8 | 104.85%
| |16384 | 1024 | 64 | 128 | 12924.74 | 60.88 | 0.85 | 51 | 14771.34 | 106.31%

### Accuracy
BF16 MLA acc:
```bash
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.9492|±  | 0.006|
|     |       |strict-match    |     5|exact_match|↑  |0.9492|±  | 0.006|
```
FP8 MLA acc:
```bash
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.9393|±  |0.0066|
|     |       |strict-match    |     5|exact_match|↑  |0.9393|±  |0.0066|
```

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

**Comment by wuhuikx** [comment]
Could you please attach the accuracy result and the performance improvement ratio?

**Comment by wuhuikx** [comment]
@ganyi1996ppo could you please help review this PR?

File: vllm/v1/attention/backends/mla/rocm_aiter_mla.py [hunk 1]
```diff
@@ -271,7 +271,7 @@ def _forward_decode(
         assert isinstance(q, torch.Tensor)
         B = q.shape[0]
         o = torch.zeros(
-            B, self.num_heads, self.kv_lora_rank, dtype=q.dtype, device=q.device
+            B, self.num_heads, self.kv_lora_rank, dtype=torch.bfloat16, device=q.device
```
Comment (ganyi1996ppo) [inline vllm/v1/attention/backends/mla/rocm_aiter_mla.py]: I wonder if we have better solution than hardcode this to bfloat16, maybe `self.model_config.dtype` ? What's your thought @tjtanaa @gshtras ?

File: vllm/v1/attention/backends/mla/rocm_aiter_mla.py [hunk 2]
```diff
@@ -55,9 +56,8 @@ class AiterMLAMetadata(MLACommonMetadata[AiterMLADecodeMetadata]):
 class AiterMLAMetadataBuilder(MLACommonMetadataBuilder[AiterMLAMetadata]):
     # TODO(luka, lucas): audit this as part of:
     #  https://github.com/vllm-project/vllm/issues/22945
-    _cudagraph_support: ClassVar[AttentionCGSupport] = (
-        AttentionCGSupport.UNIFORM_SINGLE_TOKEN_DECODE
-    )
+    cudagraph_support: ClassVar[AttentionCGSupport] = AttentionCGSupport.UNIFORM_BATCH
+    query_len_support: ClassVar[QueryLenSupport] = QueryLenSupport.VARLEN
```
Comment (ganyi1996ppo) [inline vllm/v1/attention/backends/mla/rocm_aiter_mla.py]: I don't think we should change those 2 class variable, `QueryLenSupport.VARLEN` and `AttentionCGSupport.UNIFORM_BATCH` are used only for MTP case, since your PR only enable MLA for fp8 inference, those changes may introduce functionality issue

Comment (ganyi1996ppo) [reply vllm/v1/attention/backends/mla/rocm_aiter_mla.py]: Even without performance improvement, I think this is still a important functionality feature for ROCm's MLA backend, and worth to merge. What do you think? @tjtanaa @gshtras

File: vllm/v1/attention/backends/mla/rocm_aiter_mla.py (hunk 1 above)
Comment (tjtanaa) [reply vllm/v1/attention/backends/mla/rocm_aiter_mla.py]: Great suggestion. It is better to set it to the model dtype.

File: vllm/v1/attention/backends/mla/rocm_aiter_mla.py (hunk 2 above)
Comment (tjtanaa) [reply vllm/v1/attention/backends/mla/rocm_aiter_mla.py]: Yes. It reduces KVCache memory requirement, we can fix the perf in coming PR.

**Comment by tjtanaa** [comment]
@gbyu-amd @ganyi1996ppo let me know when this is ready. We can still merge this feature as functionality enablement. The kvcache memory saving is still a benefit.

**Review by tjtanaa — APPROVED** [review]
LGTM
